rlclientlib/ranking_event.cc

```diff
@@ -51,108 +54,92 @@ namespace reinforcement_learning {
   ranking_event::ranking_event()
   { }
 
-  ranking_event::ranking_event(const char* event_id, float pass_prob, const std::string& body)
+  ranking_event::ranking_event(const char* event_id, bool deferred_action, float pass_prob, const char* context, const ranking_response& response)
     : event(event_id, pass_prob)
-    , _body(body)
-  { }
-
-  ranking_event::ranking_event(ranking_event&& other)
-    : event(std::move(other))
-    , _body(std::move(other._body))
-  {}
-
-  ranking_event& ranking_event::operator=(ranking_event&& other) {
-    if (&other != this) {
-      event::operator=(std::move(other));
-      _body = std::move(other._body);
+    , _deferred_action(deferred_action)
+    , _model_id(response.get_model_id())
+  {
+    for (auto const &r : response) {
+      _action_ids_vector.push_back(r.action_id + 1);
+      _probilities_vector.push_back(r.probability);
     }
-    return *this;
-  }
 
-  void ranking_event::serialize(u::data_buffer& oss) {
-    oss << _body;
-    if (_pass_prob < 1) {
-      oss << R"(,"pdrop":)" << (1 - _pass_prob);
-    }
-    oss << R"(})";
+    string context_str(context);
+    copy(context_str.begin(), context_str.end(), std::back_inserter(_context));
+    _approx_size = _context.size() +
+      _action_ids_vector.size() * sizeof(uint64_t) +
```

Comment:
Prefer to get these types from the vector itself
`decltype(_action_ids_vector)::value_type`

Reply:
Makes sense.

Reply:
Dead code.  No longer needed.